rinse/core.py

```diff
@@ -231,5 +303,162 @@ def __init__(self):
+        with open(filepath, "wb") as f:
+            try:
+                self.logger.info("Downloading %s" % filename)
+                response = requests.get(url, stream=True)
+                total_length = response.headers.get('content-length')
+
+                if not total_length:  # no content length header
+                    f.write(response.content)
+                else:
+                    dl = 0
+                    total_length = int(total_length)
+                    for data in response.iter_content(chunk_size=4096):
+                        dl += len(data)
+                        f.write(data)
+                        done = int(100 * dl / total_length)
+                        half_done = int(done/2)
+                        sys.stdout.write("\r[%s%s] %d%% complete" % ('=' * half_done, ' ' * (50-half_done), done))
+                        sys.stdout.flush()
+
+            except KeyboardInterrupt:
+                f.close()
+                self.logger.error("\n Ctrl-c pressed. Aborting!")
+                delete_cmd = "del %s" % filepath
+                system_cmd(cmd=delete_cmd, stdout=sp.PIPE, stderr=sp.STDOUT, shell=True)
+                sys.exit(0)
+
+        sys.stdout.write('\n')
+        self.logger.info("\n%s Downloaded Successfully" % filename)
+
+    def _url_setup(self):
+        if self.version == "latest":
+            ver = self.get_versions()[0]
+            self.version = ver
+            url = "https://cloud.r-project.org/bin/windows/base/old/%s/R-%s-win.exe" % (ver, ver)
+            filename = "R-%s-win.exe" % ver
+        else:
+            ver = self.version
+            url = "https://cloud.r-project.org/bin/windows/base/old/%s/R-%s-win.exe" % (ver, ver)
+            filename = "R-%s-win.exe" % ver
+        return url, filename
+
+
+    def _install_exe(self, exe_file_path):
+        # Check the temp directory if necessary
+        self.clear_tmp_dir()
+        # Run the R exe silently
+        cmd = '%s /VERYSILENT /DIR=%s' % (exe_file_path, self.tmp_path)
+        self.logger.info("Installing R in %s" % exe_file_path)
+        system_cmd(cmd=cmd, stdout=sp.PIPE, stderr=sp.STDOUT, shell=True)
+
+
+    def source_download(self, overwrite, with_rtools):
+        # Download the source exe
+        url, file_name = self._url_setup()
+        self.src_file_path = self.src_path / "cran" / Path(file_name)
+        src_file_path = self.src_file_path
+
+        if (not self.src_file_path.exists()) or overwrite:
+            self._url_download(url=url, filepath=self.src_file_path, 
+                               filename=file_name)
+            
+        if with_rtools:
+            self.setup_rtools()
+        return src_file_path
+    
+    def installer(self):
+        if self.method == "source":
+            src_file_path, rtools_path = self.source_download()
+            self.source_setup()
+            self.create_rhome()
+        elif self.method == "local":
+            self.use_local()
```

Comment:
We could probably remove this function since it isn't being used.

Reply:
I'm not sure about this..

Reply:
The below code was implemented in `rinse.py` that made this obsolete.

https://github.com/datasnakes/rinse/blob/2d65941de1d163970236b083480a338de764f249/rinse/rinse.py#L104-L122